tests/trainer/test_trainer.py

```diff
@@ -110,17 +110,21 @@ class TestTrainerEquivalence():
     reference_folder: pathlib.Path
 
     def assert_models_equal(self, model_1, model_2):
+        assert id(model_1) != id(model_2), "Same model should not be compared."
         for param1, param2 in zip(model_1.parameters(), model_2.parameters()):
             torch.testing.assert_allclose(param1, param2)
 
-    @pytest.fixture
+    @pytest.fixture(scope="function")
```

Comment:
this should not be neccesary, i can revert and test

Reply:
Yes, double-checked the docs and it seems that function scope should be the default.